docs_src/microservices/device/V2Migration.md

```diff
@@ -0,0 +1,85 @@
+# V2 Migration Guide
+
+!!! edgey "EdgeX 2.0"
+For the EdgeX 2.0 (Ireland) release there are many backward breaking changes. These changes require custom Device Services and custom device profiles to be migrated. This section outlines the necessary steps for this migration.
+
+## Custom Device Services
+
+### Configuration
+
+The migration of any Device Service's configuration starts with migrating configuration common to all EdgeX services. See the [V2 Migration of Common Configuration](../../configuration/V2MigrationCommonConfig) section for details. The remainder of this section focuses on configuration specific to Device Services.
+
+#### Device
+1. Remove `ImitCmd`, `ImitCmdArgs`, `RemoveCmd` and `RemoveCmdArgs`
+2. Add `UseMessageBus` to determine events should be published to MessageBus or sent by REST call.
+3. Add `DevicesDir` and `ProfilesDir` as an indication of where to load the device profiles and pre-defined devices. Convention is to put them under `/res` folder:
+```
+(where binary resides ...)
++- res
+|  +- devices
+|    +- device1.toml
+|    +- device2.toml
+|  +- profiles
+|    +- profile1.yml
+|    +- profile2.yml
+|  +- configuration.toml
+|  +- ...
++- main.go
+
+```
+
+#### MessageQueue
+See the [Device Service MessageQueue](../../design/adr/013-Device-Service-Events-Message-Bus.md#device-services) section for details
```

Comment:
Add that this is new for EdgeX 2.0. I am ok with referencing the ADR for more details, but not for the complete migration. Please add description and example TOML.